### ISSUE TITLE
DeprecationWarning in tests

### ISSUE THREAD

**Issue by StasDeep** [scrapy/scrapy#3838]
I'm using `unittest` to create tests for my spiders. Internally, I use `CrawlerRunner` to invoke callbacks.

But every time I launch tests, I get several deprecation warnings.

```
/usr/local/lib/python3.6/site-packages/scrapy/utils/conf.py:97: DeprecationWarning: The SafeConfigParser class has been renamed to ConfigParser in Python 3.2. This alias will be removed in future versions. Use ConfigParser directly instead.
  cfg = SafeConfigParser()
```

```
/usr/local/lib/python3.6/site-packages/scrapy/core/downloader/webclient.py:4: DeprecationWarning: twisted.web.client.HTTPClientFactory was deprecated in Twisted 16.7.0: please use https://pypi.org/project/treq/ or twisted.web.client.Agent instead
  from twisted.web.client import HTTPClientFactory
```

```
/usr/local/lib/python3.6/site-packages/scrapy/core/downloader/contextfactory.py:51: DeprecationWarning: Passing method to twisted.internet.ssl.CertificateOptions was deprecated in Twisted 17.1.0. Please use a combination of insecurelyLowerMinimumTo, raiseMinimumTo, and lowerMaximumSecurityTo instead, as Twisted will correctly configure the method.
  acceptableCiphers=DEFAULT_CIPHERS)
```

However, I don't see any of these warning when using any of the `scrapy` commands.

**Comment by kmike** [comment]
Some of these warnings should be fixed in #3877.

**Comment by elacuesta** [comment]
I think this is a duplicate of #3288

**Comment by Gallaecio** [comment]
Let’s close this since #3877 fixes the first warning and #3288 covers the other two already.
